src/test/java/jp/ikedam/jenkins/plugins/extensible_choice_parameter/ExtensibleChoiceParameterDefinitionJenkinsTest.java

```diff
@@ -196,10 +197,37 @@ public String getDisplayName()
             {
                 return "MockChoiceListProvider";
             }
-            
+
         }
     }
-    
+
+    public static class MockGlobalTextAreaChoiceListProvider extends GlobalTextareaChoiceListProvider

```

Comment:
Please add a new patameter to `MockChoiceListProvider` instead.
Creating a new constructor and preserving the existing constructor lets leaving other test codes unchanged. Or you can add a new setter.

Reply:
Why `MockChoiceListProvider`? 
This is not inherit from MockChoiceListProvider. It as an abstraction from `GlobalTextareaChoiceListProvider` to test that this needs BUILD permission (to prevent anyone Override the `requiresBuildPermission` in `AddEditedChoiceListProvider`.
Or I remove the test if it is unnecessary

Reply:
It's not so good idea to have tests in this file depend on actual `ChoiceListProvider`s as tests may be affected with implementation of those providers.
Tests for `AddEditedChoiceListProvider` are preferred to be written in `AddEditedChoiceListProviderTest` (not available now) if they are necessary. But I don't think it's necessary.
Anyway, overriding `GlobalTextareaChoiceListProvider` is too complicated for what we want to test. Extending `MockChoiceListProvider` is easier and simpler.

So what I want are:
* Add a parameter to `MockChoicrListProvider` to switch `requiresBuildPermission`
* Remove the test with `GlobalTextareaChoiceListProvider`.
* Use `MockChoicrListProvider` instead of `TextareaChoiceListProvider`.

Reply:
Ok
I change it on sunday